### ISSUE TITLE
IPv6 config for ingester block is not consistent since 2.9.0

### ISSUE THREAD

**Issue by mathias-ioki** [grafana/mimir#5481]
#### Describe the bug
We had mimir 2.8.0 running, with the following config (just a part of it):

```
server:
  http_listen_address: "[1f48:0e6e:e637:f::320]"

distributor:
  ring:
    kvstore:
      store: memberlist
    instance_addr: "[1f48:0e6e:e637:f::320]"

ingester:
  ring:
    instance_addr: "[1f48:0e6e:e637:f::320]"
```

With the update to 2.9.0 the instance_addr for the ingester (and only for the ingester) expect the IPv6 address without brackets. So the following config is valid for 2.9.0:

```
server:
  http_listen_address: "[1f48:0e6e:e637:f::320]"

distributor:
  ring:
    kvstore:
      store: memberlist
    instance_addr: "[1f48:0e6e:e637:f::320]"

ingester:
  ring:
    instance_addr: "1f48:0e6e:e637:f::320"
```

The 2.8.0 config results in the following error:

```
ts=2023-07-12T06:48:50.030706183Z caller=pool.go:193 level=warn msg="removing ingester failing healthcheck" addr=[[1f48:0e6e:e637:f::320]]:9095 reason="rpc error: code = Unavailable desc = connection error: desc = \"transport: Error while dialing dial tcp: address [[1f48:0e6e:e637:f::320]]:9095: missing port in address\""
```

The status page shows the double brackets, too.

#### Expected behavior

I would expect, that the IPv6 config is consistent for all configuration blocks.

#### Environment

- Infrastructure: Cloud VMs with Read-Write mode
- Deployment tool: apt packages

**Comment by dimitarvdimitrov** [comment]
This looks related to https://github.com/grafana/dskit/pull/185

**Comment by narqo** [comment]
I looked through the code, and it seems to me, it has been fully resolved with these changes: #4833 #6068 #6311. All "address" settings now expect a literal IP address (for example `192.0.2.1` or `2001:db8::2:1`).

Note: from what I can tell from the code's history, the previous support for a IPv6 form `[1f48:0e6e:e637:f::320]` in any of the `*.address` or `*.addr` settings was a bug.
